pkg/k8s/synced/crd.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package watchers
+package synced
```

Comment:
I think the refactor is good. I think a package name like `resources` would be more apt to represent the code inside it. What do you think?

Reply:
My reading of "resources" has a connotation that it would include definition or more of the resource management... This is just a simple tracker for the initial resource sync. Started with `sync` as the package name, but soon figured that was pretty bad :-)

Reply:
Fair enough.

> This is just a simple tracker for the initial resource sync

I think this is a good start to a package-level godoc. Could you create a `doc.go` and include a godoc including the above quoted text. I agree, naming this is kinda hard, so having a godoc to explain it will be valuable. 

```
Package synced provides a simple tracker to ensure that Cilium K8s resources such as CRDs are sycrhorized in the cluster. It also ... blah blah (if there's a need to expand more...)
```

Reply:
I'll leave this as-is for now, but don't oppose potential future changes for a better name.